driver_test.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2019 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License included
+// in the file licenses/BSL.txt and at www.mariadb.com/bsl11.
+//
+// Change Date: 2022-10-01
+//
+// On the date above, in accordance with the Business Source License, use
+// of this software will be governed by the Apache License, Version 2.0,
+// included in the file licenses/APL.txt and at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package datadriven
+
+import (
+	"errors"
+	"fmt"
+	"testing"
+)
+
+type A struct {
+	N int
+}
+type B struct {
+	S string
+}
+
+func TestReflectCall(t *testing.T) {
+	isEven := func(i int) bool { return i%2 == 0 }
+	popEven := func(in, exp interface{}) error {
+		*in.(*int)++
+		*exp.(*bool) = false
+
+		return nil
+	}
+
+	aToB := func(a A) B { return B{fmt.Sprintf("%03d", a.N)} }
+	popAToB := func(in, exp interface{}) error {
+		in.(*A).N = 12
+		exp.(*B).S = "foo" // not actually what aToB will produce
+		return nil
+	}
+
+	panicsOnZero := func(i int) int {
+		if i == 0 {
+			panic("boom")
+		}
+		return i
+	}
+
+	m := DriverMap{
+		"even": isEven,
+		"atob": aToB,
+		"errs": panicsOnZero,
+	}
+
+	// Two standard examples to sanity check that things "work".
+	{
```

Comment:
Consider moving each of these cases to subtests. That will make it marginally easier to debug in case they ever fail. Or not.